src/internal/_dispatchToMapMethod.js

```diff
@@ -0,0 +1,18 @@
+var _mapFunctions = require('./_mapFunctions');
+var _objectFunctions = require('./_objectFunctions');
+
+
+module.exports = (function() {
+  return typeof Map === 'function' ?
+    function _dispatchToMapMethod(methodname) {
+      return function() {
+        var isMap = arguments[arguments.length - 1] instanceof Map;
+        return (isMap ? _mapFunctions : _objectFunctions)[methodname].apply(null, arguments);
+      };
+    } :
+    function _callObjectMethod(methodname) {
+      return function() {
+        return _objectFunctions[methodname].apply(null, arguments);
```

Comment:
can't we just return a reference to the appropriate function, rather than wrapping in a lambda? this would also preserve arity which would be nice